exporters/etw/include/opentelemetry/exporters/etw/etw_properties.h

```diff
@@ -0,0 +1,476 @@
+// Copyright 2021, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#pragma once
+
+#include "opentelemetry/version.h"
+
+#include "opentelemetry/common/key_value_iterable_view.h"
+
+#include <opentelemetry/nostd/span.h>
+#include <map>
+#include <string>
+#include <vector>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace core
+{};
+namespace trace
+{};
+namespace exporter
+{
+namespace ETW
+{
+
+/**
+ * @brief PropertyVariant provides:
+ * - a constructor to initialize from initializer lists
+ * - an owning wrapper around `common::AttributeValue`
+ */
+using PropertyVariant =
+    nostd::variant<bool,
+                   int32_t,
+                   int64_t,
+                   uint32_t,
+                   uint64_t,
+                   double,
+                   std::string,
+#ifdef HAVE_CSTRING_TYPE
+                   const char *,
+#endif
+#ifdef HAVE_SPAN_BYTE
+                   // TODO: 8-bit byte arrays / binary blobs are not part of OT spec yet!
+                   // Ref: https://github.com/open-telemetry/opentelemetry-specification/issues/780
+                   std::vector<uint8_t>,
+#endif
+                   std::vector<bool>,
+                   std::vector<int32_t>,
+                   std::vector<int64_t>,
+                   std::vector<uint32_t>,
+                   std::vector<uint64_t>,
+                   std::vector<double>,
+                   std::vector<std::string>>;
+
+/**
+ * @brief PropertyValue class that holds PropertyVariant and
+ * provides converter for non-owning common::AttributeValue
+ */
+class PropertyValue : public PropertyVariant
+{
+
+  /**
+   * @brief Convert span<T> to vector<T>
+   * @tparam T
+   * @param source
+   * @return
+   */
+  template <typename T>
+  static std::vector<T> to_vector(const nostd::span<const T, std::size_t(-1)> &source)
```

Comment:
Prefer `std::numeric_limits<std::size_t>::max()` over `-1`?

Reply:
I remembered one thing about `max` here : it's by default broken, since `NOMINMAX` macro isn't default.
The old issue: http://www.suodenjoki.dk/us/archive/2010/min-max.htm

The pattern here is identical to implementation of `dynamic extent` in GSL from here:
https://github.com/microsoft/GSL/blob/ef0ffefe525a6219ff245d19a832ce06f3fd3504/include/gsl/span#L64

I think it'd probably be more prudent to add a similar definition of `dynamic_extent` constant to `nostd::span` instead.

Reply:
Basically Cpp reference suggests:
https://en.cppreference.com/w/cpp/container/span/dynamic_extent

```cpp
inline constexpr std::size_t dynamic_extent = -1;
```
which I think we should be adding to `nostd::` namespace.

Reply:
Ok, fixed in the next iteration.